pkg/storage/tsdb/config.go

```diff
@@ -108,6 +112,8 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 	f.IntVar(&cfg.ShipConcurrency, "experimental.tsdb.ship-concurrency", 10, "Maximum number of tenants concurrently shipping blocks to the storage.")
 	f.StringVar(&cfg.Backend, "experimental.tsdb.backend", "s3", "TSDB storage backend to use")
 	f.IntVar(&cfg.MaxTSDBOpeningConcurrencyOnStartup, "experimental.tsdb.max-tsdb-opening-concurrency-on-startup", 10, "limit the number of concurrently opening TSDB's on startup")
+	f.DurationVar(&cfg.HeadCompactionInterval, "experimental.tsdb.head-compaction-interval", 5*time.Minute, "How frequently does Cortex try to compact TSDB head. Block is only created if data covers smallest block range.")
```

Comment:
5m looks a bit unfrequent to me. This is a very inexpensive operation and I would suggest to run it more frequently: 30s or 1m.

Reply:
Do we even need to make it configurable? I'd just hardcode 1m into the code.

Reply:
I thought about hardcoding to 1m as well, and that was my initial comment. I then changed my mind because I think being able to configure it could help future integration tests (ie. compact every 5s to have integration tests running faster). What's your take?

P.S. I'm talking about integration tests cause I would like to add at least 1 integration test testing the compaction and shipping soon.

Reply:
We need a secret option then ;-) I'd suggest adding a check to avoid setting it too high (> 5 mins).

Reply:
> We need a secret option then ;-)

We can have a secret YAML config option with `doc:"hidden"` but I'm not sure it's possible to have an hidden CLI flag (I haven't investigated).